### DISCUSSION TITLE
Notification controller returns HTTP 429 for webhook events from github

### DISCUSSION THREAD

**Discussion by tobias-jenkner** [fluxcd/flux2#3571] [Q&A]
We have a problem with github webhook events. Github sometimes receives a HTTP 429 (too many requests) return code from the notification controller when sending a webhook event, which we do not understand. 

Does anybody have an idea why this happens ?

**Comment by somtochiama** [comment]
The http servers in the notification controller are rate limited. How frequently is Github sending the webhook events?

**Reply by stefanprodan** [reply]
We don't rate limit webhook receivers, but only our own events. The rate limit probably is made by the load balancer that sits in front of notification-controller.

**Reply by tobias-jenkner** [reply]
@stefanprodan If I send a webhook event twice manually with curl from a pod running in the cluster I receive http 429, meaning it is rate limited, isn't it? 

> curl -X POST --data @payload.json -H @headers.txt -v http://notification-controller.flux-system.svc.cluster.local

Do I do anything wrong here?

**Reply by stefanprodan** [reply]
The webhook receiver is exposed as `webhook-receiver` https://github.com/fluxcd/notification-controller/blob/main/config/manager/service.yaml#L20

**Reply by tobias-jenkner** [reply]
I guess that means that we experience rate limiting , because we use the notification-controller service instead of the webhook-receiver service, is that correct?

**Reply by stefanprodan** [reply]
Yes, please see the docs here https://fluxcd.io/flux/components/notification/receiver/#public-ingress-considerations

**Comment by stefanprodan** [comment]
@tobias-jenkner how are you exposing notification-controller on the internet? Do you use an ingress controller or a load balancer? You may want to look into those and increase the rate limits there.

**Reply by tobias-jenkner** [reply]
We are using https://kyma-project.io/ and are therefore exposing the notification-controller through an API-Rule (see https://github.com/kyma-project/api-gateway) on the internet. Our cloud provider that is responsible for the kyma setup already assured us that there is no rate limitting happening in the kyma-api-gateway by default.
